packages/compass-saved-aggregations-queries/src/components/aggregations-queries-list.tsx

```diff
@@ -1,99 +1,131 @@
+import type { Item } from '../stores/aggregations-queries-items';
 import { openSavedItem } from '../stores/open-item';
 import type { RootActions, RootState } from '../stores/index';
 import { SavedItemCard, CARD_WIDTH, CARD_HEIGHT } from './saved-item-card';
 import type { SavedItemCardProps, Action } from './saved-item-card';
 import OpenItemModal from './open-item-modal';
+import { useGridFilters, useFilteredItems } from '../hooks/use-grid-filters';
 
-const ConnectedItemCard = connect<
-  Omit<SavedItemCardProps, 'onAction'>,
-  Pick<SavedItemCardProps, 'onAction'>,
-  { index: number },
-  RootState
->(
-  ({ savedItems: { items } }, props) => {
-    const item = items[props.index];
-
-    return {
-      id: item.id,
-      type: item.type,
-      name: item.name,
-      database: item.database,
-      collection: item.collection,
-      lastModified: item.lastModified,
-    };
+type SortKeys = Exclude<
+  Extract<keyof Item, string>,
+  'id' | 'database' | 'collection' | 'type'
+>;
+
+const sortBy: { name: SortKeys; label: string }[] = [
+  {
+    name: 'name',
+    label: 'Name',
   },
   {
-    onAction(id: string, actionName: Action) {
-      return (
-        dispatch: ThunkDispatch<RootState, void, RootActions>,
-        getState: () => RootState
-      ) => {
-        switch (actionName) {
-          case 'open':
-            return openSavedItem(id)(dispatch, getState);
-        }
-      };
-    },
-  }
-)(SavedItemCard);
+    name: 'lastModified',
+    label: 'Last Modified',
+  },
+];
 
-const header = css({
+const headerStyles = css({
   margin: spacing[3],
+  display: 'flex',
+  justifyContent: 'space-between',
 });
 
-const title = css({
-  marginBottom: spacing[1],
-});
-
-const row = css({
+const rowStyles = css({
   gap: spacing[2],
   paddingLeft: spacing[3],
   paddingRight: spacing[3],
   paddingBottom: spacing[2],
 });
 
+const ControlsContext = React.createContext<{
+  filterControls: React.ReactElement | null;
+  sortControls: React.ReactElement | null;
+}>({
+  filterControls: null,
+  sortControls: null,
+});
+
+const GridControls = () => {
+  const { filterControls, sortControls } = useContext(ControlsContext);
+
+  return (
+    <div className={headerStyles}>
+      <div>{filterControls}</div>
+      <div>{sortControls}</div>
+    </div>
+  );
+};
+
 const AggregationsQueriesList = ({
   loading,
   items,
   fetchItems,
+  onAction,
 }: AggregationsQueriesListProps) => {
   useEffect(() => {
     void fetchItems();
   }, [fetchItems]);
 
+  const [filterControls, filters, search] = useGridFilters(items);
+  const filteredItems = useFilteredItems(items, filters, search);
+  // If a user is searching, we disable the sort as
+  // search results are sorted by match score
+  const isDisabled = Boolean(search) ?? false;
+  const [sortControls, sortState] = useSortControls<SortKeys>(sortBy, {
+    isDisabled,
+  });
+  const sortedItems = useSortedItems(
+    filteredItems,
+    isDisabled ? null : sortState
+  );
```

Comment:
I think we should keep the two concepts separate, you are currently mixing filter with sort and then have to disable sort controls based on the filter values making the interface of `useSortedItems` confusing and requiring to not pass sortState to it. I think if this requires such a special handling here, we probably should avoid moving this hook to shared and generic compass-components

Reply:
as discussed, moved this check to `useSortControls` to return `sortState` conditionally.